server/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreRepository.java

```diff
@@ -457,12 +503,16 @@ private void cleanupStaleRootFiles(Set<String> rootBlobNames, RepositoryData rep
                 return false;
             }
         ).collect(Collectors.toList());
+    }
+
+    private List<String> cleanupStaleRootFiles(List<String> blobsToDelete) {
```

Comment:
@original-brownbear do you think we need to add the same logic to the cleanup tool?

Reply:
We could add it, I'd expect the effects to be somewhat minor though (unless there is an unknown bug that leads to thousands of leaked files here that add listing costs or a huge cluster state is snapshotted repeatedly) for most repos. But come to think of it ... we had some buggy ones that would never finalise correctly but go through all the other snapshot steps.  I guess I'd check if this is an issue first, then decide if it's worth the added risk?